azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -180,6 +185,70 @@ def disable(self):
             self._cgroups_enabled = False
             CGroupsTelemetry.reset()
 
+        @staticmethod
+        def __collect_azure_unit_telemetry():
+            azure_unit_name = None
+            azure_unit_description = None
+            try:
+                units = shellutil.run_command(['systemctl', 'list-units', 'azure*', '-all'])
+                for line in units.split('\n'):
+                    match = re.match(r'\s?(azure[^\s]*)\s?', line, re.IGNORECASE)
+                    if match is not None:
+                        azure_unit_name = match.group(1)
+                        azure_unit_description = line
```

Comment:
Shouldn't you break once you find your first match? Or maybe capture all the possible options (and make `azure_unit_name` an array instead?)

Reply:
good point; i'll make it handle multiple instances (if any)

Reply:
done